test/e2e/util.go

```diff
@@ -150,3 +152,196 @@ func RunJobScript(t *testing.T, clients *test.Clients, image, script string, sel
 		t.Fatalf("Error waiting for Job to complete successfully: %v", waitErr)
 	}
 }
+
+func RunJobListener(t *testing.T, clients *test.Clients) (string, context.CancelFunc, context.CancelFunc) {
+	name := helpers.ObjectNameForTest(t)
+
+	selector := map[string]string{
+		"job-name": name,
+	}
+
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: test.Namespace,
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{{
+						Name:            "listener",
+						Image:           pkgtest.ImagePath("listener"),
+						ImagePullPolicy: corev1.PullIfNotPresent,
+						Ports: []corev1.ContainerPort{{
+							Name:          "http",
+							ContainerPort: 8080,
+						}},
+						Env: []corev1.EnvVar{{
+							Name:  "PORT",
+							Value: "8080",
+						}},
+						ReadinessProbe: &corev1.Probe{
+							Handler: corev1.Handler{
+								TCPSocket: &corev1.TCPSocketAction{
+									Port: intstr.FromInt(8080),
+								},
+							},
+						},
+					}},
+					RestartPolicy: corev1.RestartPolicyNever,
+				},
+			},
+		},
+	}
+	pkgtest.CleanupOnInterrupt(func() {
+		clients.KubeClient.Kube.BatchV1().Jobs(job.Namespace).Delete(job.Name, &metav1.DeleteOptions{})
+	}, t.Logf)
+	job, err := clients.KubeClient.Kube.BatchV1().Jobs(job.Namespace).Create(job)
+	if err != nil {
+		t.Fatalf("Error creating Job: %v", err)
+	}
+
+	// Dump the state of the Job after it's been created so that we can
+	// see the effects of the binding for debugging.
+	t.Log("", "job", spew.Sprint(job))
+
+	cancel := func() {
+		err := clients.KubeClient.Kube.BatchV1().Jobs(job.Namespace).Delete(job.Name, &metav1.DeleteOptions{})
+		if err != nil {
+			t.Errorf("Error cleaning up Job %s", job.Name)
+		}
+	}
+	waiter := func() {
+		// Wait for the Pod to show up in the Endpoints resource.
```

Comment:
this comment is wrong? Left over cut&paste?

Reply:
good catch.